api/v1alpha1/dnsrecord_types.go

```diff
@@ -33,10 +33,19 @@ const HttpsProtocol HealthProtocol = "HTTPS"
 // By default this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint         string          `json:"endpoint,omitempty"`
-	Port             *int            `json:"port,omitempty"`
-	Protocol         *HealthProtocol `json:"protocol,omitempty"`
-	FailureThreshold *int            `json:"failureThreshold,omitempty"`
+	// Should start with "?" or "|", contain only allowed for URLs characters and end with alphanumeric char or "/"
+	// +kubebuilder:validation:Pattern=`^(?:\?|\/)[\w\-.~:\/?#\[\]@!$&'()*+,;=]+(?:[a-zA-Z0-9]|\/){1}$`
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// Only following ports allowed: 80, 443, 1024-49151
+	// +kubebuilder:validation:XValidation:rule="self in [80, 443] || (self >= 1024 && self <= 49151)",message="Allow ports 80, 443, 1024-49151 are allowed"
+	Port *int `json:"port,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self in ['HTTP','HTTPS']",message="Only HTTP or HTTPS protocols are allowed"
+	Protocol *HealthProtocol `json:"protocol,omitempty"`
```

Comment:
Using Enum would probably make more sense here, and for port see https://github.com/Kuadrant/kuadrant-operator/blob/main/api/v1alpha1/dnspolicy_types.go#L255

Also, all the validations should be the same as https://github.com/Kuadrant/kuadrant-operator/blob/main/api/v1alpha1/dnspolicy_types.go#L246-L261

@philbrookes Why is that duplicated? Would it not make more sense for the DNSPolicy to reference this HealthCheckSpec type?

Reply:
I cannot remember clearly, I suspect the Specs were different from each other at some point, but are aligned now.

Reply:
as the validations in this PR are superior to the validations in Kuadrant-operator, it might be worth a PR to update kuadrant to use this.

Reply:
I'd prefer the CEL validations - way nicer error message if things aren't right. 

Hmm, would it be a good idea to remove duplication as part of this issue?

Reply:
Opened a [PR](https://github.com/Kuadrant/kuadrant-operator/pull/723) to the kuadrant-operator as well